### PR TITLE
Create collapsing table for cohortpeople

### PR DESCRIPTION
**Pull request by EDsCODE**
## Changes

*Please describe.*  
- separate table migration from #4428 for deployment purposes
*If this affects the frontend, include screenshots.*  

## Checklist

- [ ] All querysets/queries filter by Organization, by Team, and by User
- [ ] Django backend tests
- [ ] Jest frontend tests
- [ ] Cypress end-to-end tests
- [ ] Migrations are safe to run at scale (e.g. PostHog Cloud) – present proof if not obvious

### CODE REVIEW

File: ee/clickhouse/sql/cohort.py [hunk 1]
```diff
@@ -1,3 +1,18 @@
...
 """
+
+CREATE_COHORTPEOPLE_TABLE_SQL = """
+CREATE TABLE cohortpeople
+(
+    person_id UUID,
+    cohort_id Int64,
+    team_id Int64,
+    sign Int8
+) ENGINE = CollapsingMergeTree(sign)
```
Comment (macobo) [inline ee/clickhouse/sql/cohort.py:12]: Problem - this needs to be `replicated` as the events/person tables are.

From ee/clickhouse/sql/clickhouse.py:
```
TABLE_ENGINE = (
    "ReplicatedReplacingMergeTree('/clickhouse/tables/{{shard}}/posthog.{table}', '{{replica}}', {ver})"
    if CLICKHOUSE_REPLICATION
    else "ReplacingMergeTree({ver})"
)
```

Comment (EDsCODE) [reply ee/clickhouse/sql/cohort.py]: yup good point

Comment (EDsCODE) [reply ee/clickhouse/sql/cohort.py]: I added storage policy too assuming it's the same as the other tables

File: ee/clickhouse/sql/clickhouse.py [hunk 2]
```diff
@@ -36,9 +42,13 @@
...
+COLLAPSING_MERGE_TREE = "collapsing_merge_tree"
+
 
-def table_engine(table: str, ver: Optional[str] = None) -> str:
-    if ver:
+def table_engine(table: str, ver: Optional[str] = None, engine_type: Optional[str] = None) -> str:
+    if engine_type == COLLAPSING_MERGE_TREE and ver:
+        return COLLAPSING_TABLE_ENGINE.format(table=table, ver=ver)
+    elif ver:
         return TABLE_ENGINE.format(table=table, ver=ver)
```
Comment (macobo) [inline ee/clickhouse/sql/clickhouse.py:52]: Perhaps worth renaming this to `REPLACING_TABLE_ENGINE` or smth + make sure we pass in engine_type in this branch as well? :)

**Review by macobo — APPROVED** [review]
Small nit, other than that lgtm. Will merge + deploy once you're also happy (comment below)

File: ee/clickhouse/sql/clickhouse.py (hunk 2 above)
Comment (EDsCODE) [reply ee/clickhouse/sql/clickhouse.py]: yup just update those and made sure engine_type is passed in everywhere

**Comment by EDsCODE** [comment]
docker build failing? but I think it's otherwise ready!

Edit: good to go

**Pull request merged**